### DISCUSSION TITLE
NextJS Production build displaying error with code

### DISCUSSION THREAD

**Discussion by dhavalsoni2001** [vercel/next.js#12956] [Help] [answered]
![Screenshot 2020-05-16 at 3 27 56 AM](https://user-images.githubusercontent.com/11704436/82099646-77571000-9725-11ea-83d1-51608e4dce3f.png)
I have deployed the next app but somehow there was some issue and I got the error log which contains the code on production.

I am on the latest NextJS version, I  am using the following command to build the app and run the app. So I doubt that it's not properly built and still behaving as development mode.
`npm run build` -> next build
`npm run start` ->  SET NODE_ENV=production & node  start-server.js

**Comment by jamesmosier** [comment]
Could you share some code? Perhaps your `start-server.js`?

**Reply by dhavalsoni2001** [reply]
@jamesmosier Below is my start-server.js which is calling `server/index.js`.  I am using express.

```
require("@babel/register")({
    presets: ["@babel/preset-env","next/babel"],
    
});  

 require('./server');
```

**Reply by jamesmosier** [reply]
This doesn't really show a lot unfortunately. Could you share a repository with this code? Or even just a minimal reproduction of it?

**Comment by dhavalsoni2001** [comment, marked as answer]
Hey @jamesmosier finally it's got fixed... earlier I was using `SET` command to set production but it's not executing properly and always run in `dev` environment.

```
  "scripts": {
    "dev": "node start-server.js",
    "build": "next build",
    "start": "SET NODE_ENV=production node start-server.js",
    }
```

Now I have replaced it with `cross-env` and it's working perfectly fine.


```
  "scripts": {
    "dev": "node start-server.js",
    "build": "next build",
    "start": "cross-env NODE_ENV=production node start-server.js",
    }
```

**Comment by aryankarim** [comment]
why can't we just say `NODE_ENV=production` or `NODE_ENV=development` in the .env.local?
